Tests/Functional/Controller/ConnectTest.php

```diff
@@ -0,0 +1,204 @@
+
+    /**
+     * @param Response $response
+     * @return array
+     */
+    private function handleJsonResponse(Response $response)
+    {
+        $this->assertEquals(200, $response->getStatusCode());
+        $responseData = \json_decode($response->getContent(), true);
+        $this->assertNotNull($responseData, 'Response is not valid JSON');
+
+        return $responseData;
+    }
+
+    public function test_get_article_source_ids()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+        $expectedIds = ['2', '2-123', '2-124'];
+
+        $client->request('POST', 'backend/Connect/getArticleSourceIds', ['ids' => [2]]);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertNotEmpty($responseData);
+        $this->assertEquals(1, $responseData['success']);
+        $this->assertEquals($expectedIds, $responseData['sourceIds']);
+    }
+
+    public function test_get_article_source_ids_with_export_all()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+        $expectedIds = ['2', '117', '2-123', '2-124', '117-221'];
+
+        $client->request('POST', 'backend/Connect/getArticleSourceIds', ['exportAll' => 'true']);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertNotEmpty($responseData);
+        $this->assertTrue($responseData['success']);
+        $this->assertArraySubset($expectedIds, $responseData['sourceIds']);
+    }
+
+    public function test_exporting_a_product_with_variants()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+
+        $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => ['2','2-123','2-124']]);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertTrue($responseData['success']);
+    }
+
+    public function test_error_handling_on_export()
+    {
+        /** @var TestClient $client */
+        $client = self::createBackendClient();
+
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
+
+        $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => ['2','2-123']]);
+        $responseData = $this->handleJsonResponse($client->getResponse());
+
+        $this->assertFalse($responseData['success']);
+        $this->assertCount(2, $responseData['messages']['price']);
```

Comment:
here ypou can check db for error messages to, and if you test is one "good" product assert the same as in the testcase above in the db